### DISCUSSION TITLE
`auth.storage` with SecureStore is overloaded with ReactNative/Expo

### DISCUSSION THREAD

**Discussion by lucksp** [supabase/supabase#14906] [Questions]
```
// as suggested in https://supabase.com/docs/guides/getting-started/tutorials/with-expo
const ExpoSecureStoreAdapter = {
  getItem: async (key = '') => {
    return await SecureStore.getItemAsync(key);
  },
  setItem: async (key = '', value = '') => {
    return await SecureStore.setItemAsync(key, value);
  },
  removeItem: async (key = '') => {
    return await SecureStore.deleteItemAsync(key);
  },
};

export const supabase = createClient(SUPABASE_MFI_URL, SUPABASE_MFI_API, {
  auth: {
    storage: ExpoSecureStoreAdapter,
    autoRefreshToken: true,
    persistSession: true,
    detectSessionInUrl: false,
  },
});
```

This example overloads the storage limit for `SecureStore`.  How important is it to use SecureStore?

**Comment by GaryAustin1** [comment]
You can use  any local storage solution for auth as long as you only use anon key on the client.

**Reply by lucksp** [reply]
I can't control what gets set by supabase via `createClient`...correct?

When you say:
>  only use anon key on the client

What do you mean?

**Comment by GaryAustin1** [comment]
Supabase is storing the user session there.  So you really can't control it and the user jwt can get large based on user and app metadata.

I guess the anon key (versus service_role) is not relevant to the storage discussion, that is more serverside only versus using on a browser client.
